### PR TITLE
role assignment updates

### PR DESCRIPTION
**Pull request by wesleymccollam**

### CODE REVIEW

File: internal/connector/pingone/platform/resources/pingone_gateway_role_assignment.go [hunk 1]
```diff
@@ -59,13 +59,13 @@ func (r *PingoneGatewayRoleAssignmentResource) ExportAll() (*[]connector.ImportB
 						return nil, err
 					}
 
-					for roleAssignmentIndex, roleAssignment := range gatewayRoleAssignmentsEmbedded.GetRoleAssignments() {
+					for _, roleAssignment := range gatewayRoleAssignmentsEmbedded.GetRoleAssignments() {
 						roleAssignmentId, roleAssignmentIdOk := roleAssignment.GetIdOk()
-
-						if roleAssignmentIdOk {
+						roleAssignmentRoleName, roleAssignmentRoleOk := roleAssignment.GetRoleOk()
```
Comment (erikostien-pingidentity) [inline internal/connector/pingone/platform/resources/pingone_gateway_role_assignment.go:64]: Does Gateway have the idea of AccessControl or does the application role assignment have the GetRoleOk() function?

Comment (erikostien-pingidentity) [reply internal/connector/pingone/platform/resources/pingone_gateway_role_assignment.go:64]: Nevermind, I see the approleassignment moved to using GetRoleOk as well

File: internal/connector/pingone/sso/resources/pingone_application_role_assignment.go [hunk 2]
```diff
@@ -90,10 +90,11 @@ func (r *PingoneApplicationRoleAssignmentResource) ExportAll() (*[]connector.Imp
 
 			for _, roleAssignment := range appRoleAssignmentsEmbedded.GetRoleAssignments() {
 				roleAssignmentId, roleAssignmentIdOk := roleAssignment.GetIdOk()
-				if roleAssignmentIdOk {
+				roleAssignmentRoleName, roleAssignmentRoleOk := roleAssignment.GetRoleOk()
```
Comment (erikostien-pingidentity) [inline internal/connector/pingone/sso/resources/pingone_application_role_assignment.go:93]: If we do this here, the above logic to check the AccessControl role is not needed

File: internal/connector/pingone/platform/resources/pingone_gateway_role_assignment.go [hunk 3]
```diff
@@ -59,13 +59,13 @@ func (r *PingoneGatewayRoleAssignmentResource) ExportAll() (*[]connector.ImportB
...
+						if roleAssignmentIdOk && roleAssignmentRoleOk {
 							importBlocks = append(importBlocks, connector.ImportBlock{
 								ResourceType: r.ResourceType(),
-								ResourceName: fmt.Sprintf("%s_role_assignment_%d", *gatewayName, (roleAssignmentIndex + 1)),
+								ResourceName: fmt.Sprintf("%s_role_assignment_%s", *gatewayName, *roleAssignmentRoleName),
```
Comment (erikostien-pingidentity) [inline internal/connector/pingone/platform/resources/pingone_gateway_role_assignment.go:68]: Either add "role_assignment" to the application ResourceName, or remove it here for parity.

File: internal/connector/pingone/sso/resources/pingone_application_role_assignment.go (hunk 2 above)
Comment (wesleymccollam) [reply internal/connector/pingone/sso/resources/pingone_application_role_assignment.go:93]: name

File: internal/connector/pingone/platform/resources/pingone_gateway_role_assignment.go (hunk 3 above)
Comment (wesleymccollam) [reply internal/connector/pingone/platform/resources/pingone_gateway_role_assignment.go]: removed

**Review by erikostien-pingidentity — APPROVED** [review]
Looks Good

**Pull request merged**